pkg/openstack/loadbalancer.go

```diff
@@ -1680,22 +1680,20 @@ func (lbaas *LbaasV2) checkService(service *corev1.Service, nodes []*corev1.Node
 			}
 		}
 
+		// If LB class doesn't define FIP network or subnet, get it from svc annotation or fall back to configuration
 		if floatingNetworkID == "" {
 			floatingNetworkID = getStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerFloatingNetworkID, lbaas.opts.FloatingNetworkID)
-			if floatingNetworkID == "" {
-				var err error
-				floatingNetworkID, err = openstackutil.GetFloatingNetworkID(lbaas.network)
-				if err != nil {
-					klog.Warningf("Failed to find floating-network-id for Service %s: %v", serviceName, err)
-				}
-			}
 		}
 
-		// apply defaults from CCM config
+		// If there's no annotation and configuration, try to autodetect the FIP network by looking up external nets
 		if floatingNetworkID == "" {
-			floatingNetworkID = lbaas.opts.FloatingNetworkID
```

Comment:
you mean this line `			floatingNetworkID = lbaas.opts.FloatingNetworkID` is dead code ?

Reply:
ok, this make sense , sorry for the late reply :)